spec.html

```diff
@@ -845,29 +845,22 @@ <h1>Syntax-Directed Operations</h1>
 
     <emu-clause id="sec-runtime-semantics">
       <h1>Runtime Semantics</h1>
-      <p>Algorithms which specify semantics that must be called at runtime are called <dfn>runtime semantics</dfn>. Runtime semantics are defined by abstract operations or syntax-directed operations. Such algorithms always return a completion record.</p>
+      <p>Algorithms which specify semantics that must be called at runtime are called <dfn>runtime semantics</dfn>. Runtime semantics are defined by abstract operations or syntax-directed operations.</p>
 
-      <emu-clause id="sec-implicit-completion-values">
-        <h1>Implicit Completion Values</h1>
-        <p>The algorithms of this specification often implicitly return Completion Records whose [[Type]] is ~normal~. Unless it is otherwise obvious from the context, an algorithm statement that returns a value that is not a Completion Record, such as:</p>
-        <emu-alg>
-          1. Return *"Infinity"*.
-        </emu-alg>
-        <p>means the same thing as:</p>
-        <emu-alg>
-          1. Return NormalCompletion(*"Infinity"*).
-        </emu-alg>
-        <p>However, if the value expression of a &ldquo;return&rdquo; statement is a Completion Record construction literal, the resulting Completion Record is returned. If the value expression is a call to an abstract operation, the &ldquo;return&rdquo; statement simply returns the Completion Record produced by the abstract operation.</p>
-        <p>The abstract operation Completion(_completionRecord_) is used to emphasize that a previously computed Completion Record is being returned. The Completion abstract operation takes a single argument, _completionRecord_, and performs the following steps:</p>
+      <emu-clause id="sec-completion-ao" type="abstract operation">
+        <h1>
+          Completion (
+            _completionRecord_: a Completion Record,
+          ): a Completion Record
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It is used to emphasize that a Completion Record is being returned.</dd>
```

Comment:
In the light of explicit emphasis via a return type, does `Completion()` give any benefit or it may be removed altogether?

Reply:
You don't have the return type sitting in front of you at the call site, only at the definition site, so yes, it's still helpful.